docs/experiments-fever.md

```diff
@@ -0,0 +1,196 @@
+ --truth_file collections/fever/paper_dev.jsonl \
+ --run_file runs/run.fever-paragraph.dev.tsv
+```
+
+This run produces the following results:
+
+| k   | Fully Supported | Oracle Accuracy |
+|:----|----------------:|----------------:|
+| 1	  | 0.3271827183    | 0.5514551455    |
+| 5	  | 0.5655565557    | 0.7103710371    |
+| 10  |	0.6542154215    | 0.7694769477    |
+| 25  |	0.7458745875    | 0.8305830583    |
+| 50  |	0.8097809781    | 0.8731873187    |
+| 100 |	0.8561356136    | 0.9040904090    |
+
+### BM25 Tuning
+
+The above retrieval uses the MS MARCO default BM25 parameters of `k1=0.82`, `b=0.68`. We can tune these parameters to outperform the results of the TF-IDF baseline in the paper.
+
+We tune on a subset of the training split of the dataset. We generate that subset:
+
+```bash
+python src/main/python/fever/generate_subset.py \
+ --dataset_file collections/fever/train.jsonl \
+ --subset_file collections/fever/train-subset.jsonl
+```
+
+We then generate the queries and qrels files for this subset.
+
+```bash
+python src/main/python/fever/generate_queries_and_qrels.py \
+ --dataset_file collections/fever/train-subset.jsonl \
+ --output_queries_file collections/fever/queries.paragraph.train-subset.tsv \
+ --output_qrels_file collections/fever/qrels.paragraph.train-subset.tsv \
+ --granularity paragraph
+```
+
+We tune the BM25 parameters with a grid search of parameter values in 0.1 increments. We save the run files generated by this process to a new folder `runs/fever-bm25` (do not use `runs` here).
+
+```bash
+python src/main/python/fever/tune_bm25.py \
+ --runs_folder runs/fever-bm25 \
+ --index_folder indexes/fever/lucene-index-fever-paragraph \
+ --queries_file collections/fever/queries.paragraph.train-subset.tsv \
+ --qrels_file collections/fever/qrels.paragraph.train-subset.tsv
+```
+
+From the grid search, we observe that the parameters `k1=0.6`, `b=0.5` perform fairly well. If we retrieve on the dev set with these parameters:
+
+```bash
+python tools/scripts/msmarco/retrieve.py \
+ --hits 1000 --threads 1 \
+ --index indexes/fever/lucene-index-fever-paragraph \
+ --queries collections/fever/queries.paragraph.dev.tsv \
+ --output runs/run.fever-paragraph-0.6-0.5.dev.tsv \
+ --k1 0.6 --b 0.5
+```
+
+and we evaluate this run file:
+
+```bash
+python src/main/python/fever/evaluate_doc_retrieval.py \
+ --truth_file collections/fever/paper_dev.jsonl \
+ --run_file runs/run.fever-paragraph-0.6-0.5.dev.tsv
+```
+
+then we can achieve the following results:
+
+| k   | Fully Supported | Oracle Accuracy |
+|:----|----------------:|----------------:|
+| 1	  | 0.3856885689    | 0.5904590459    |
+| 5	  | 0.6366636664    | 0.7577757776    |
+| 10  | 0.7193219322    | 0.8128812881    |
+| 25  | 0.8003300330    | 0.8668866887    |
+| 50  | 0.8472847285    | 0.8981898190    |
+| 100 | 0.8804380438    | 0.9202920292    |
+
+which outperforms the TF-IDF baseline in the FEVER paper at almost every value of k.
```

Comment:
```suggestion
which outperforms the TF-IDF baseline in the FEVER paper at every tested value of k.
```

Reply:
Unfortunately we don't seem to outperform TF-IDF at k=50?

Reply:
Oh nvm, looks like their reported oracle accuracy is wrong (assuming their fully supported % is right)

Reply:
yeah, I just looked at the first column since the second is determined exclusively by it.